lib/date/date.ex

```diff
@@ -1298,4 +1298,18 @@ defmodule Timex.Date do
     {d,{h,m,s,round(u/1000)}}
   end
 
+  @doc """
+  Set the date to the beginning of the year
```

Comment:
How often do people want to provide a DateTime and get a completely different one back (in this case representing the beginning of the year)? It seems to me that these should be functions in the vein of `Date.iso_week`, etc., which produce a DateTime from some value. In this case, there is one required value (year), and one optional value (the timezone). These functions are more useful if they can be the starting point for getting a DateTime than require you to already have one, since if you already have a DateTime, it's trivial to extract the parts you need to call this function anyway. Thoughts?